src/normal.c

```diff
@@ -346,6 +347,72 @@ static const struct nv_cmd
     {K_PS, nv_edit, 0, 0},
 };
 
+#define strstartswith(a, b) (!strncmp(a, b, strlen(b)))
+
+void toggle_comment(linenr_T lnum)
+{
+  const char_u *comment = curbuf->b_oni_line_comment != NULL ? curbuf->b_oni_line_comment : (char_u *)"//";
+  int commentlen = (int)STRLEN(comment);
+  const char_u *line = ml_get(lnum);
+  int linelen = (int)STRLEN(line);
+  char_u *newp;
+
+  if (strstartswith(line, comment))
+  {
+    // remove comment
+
+    newp = alloc((linelen - commentlen) + 1);
+
+    if (newp == NULL)
+      return;
+
+    if (virtual_active() && curwin->w_cursor.coladd > 0)
+      coladvance_force(getviscol());
+
+    mch_memmove(newp, line + commentlen, (size_t)((linelen - commentlen) + 1));
+    ml_replace(lnum, newp, FALSE);
```

Comment:
I tested the latest and it seems we don't get a buffer update in the 'remove' case. I think we need to call `del_bytes` or `del_chars` to trigger the buffer update.

Reply:
Alternatively - instead of calling `inserted_bytes`, we could call `changed_lines` in `toggle_comment_lines` - that should trigger a buffer update, for both cases.

Reply:
I could also have called `inserted_bytes` with a negative count, but calling `changed_lines` once seems better to avoid unnecessary updates.

FIxed in https://github.com/onivim/libvim/pull/188/commits/bca12bff67a00349d311c044b38979072292b0ed